scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+         --initialize-insecure --wsrep_provider=none --datadir="$tmp_datadir" >> $CLONE_ERR 2>&1 || \
+    ( wsrep_log_error "Failed to initialize data directory. Some hints below:"
+      grep '[ERROR]' $CLONE_ERR | cut -d ']' -f 4- | while read msg
+      do
+          wsrep_log_error "> $msg"
+      done
+      wsrep_log_error "Full log at $CLONE_ERR"
+      exit 1 )
+
+    # Move initialized data directory structure to real datadir and cleanup
+    mv --force "$tmp_datadir"/* "$WSREP_SST_OPT_DATA/"
+    sleep 2
+    wsrep_log_debug "-> REMOVE $tmp_datadir"
+    rm -rf "$tmp_datadir"
+
+    wsrep_log_info "INITIALIZE DB DONE"
+
+    # Need to create an extra user for 'localhost' because in some installations
+    # by default exists user ''@'localhost' and shadows every user with '%'.
+ cat << EOF > "$CLONE_SQL"
+ SET SESSION sql_log_bin=OFF;
+ CREATE USER '$CLONE_USER'@'%' IDENTIFIED $WITH_OPTION BY '$CLONE_PSWD';
+ GRANT CLONE_ADMIN, SYSTEM_VARIABLES_ADMIN, SUPER, SHUTDOWN, EXECUTE ON *.* to '$CLONE_USER'@'%';
+ GRANT INSERT ON mysql.plugin to '$CLONE_USER';
+ GRANT SELECT,UPDATE,INSERT ON performance_schema.* TO '$CLONE_USER';
+ CREATE USER '$CLONE_USER'@'localhost' IDENTIFIED $WITH_OPTION BY '$CLONE_PSWD';
+ GRANT CLONE_ADMIN, SYSTEM_VARIABLES_ADMIN, SUPER, SHUTDOWN, EXECUTE ON *.* TO '$CLONE_USER'@'localhost';
+ GRANT EXECUTE ON *.* to '$CLONE_USER'@'localhost';
+ GRANT INSERT ON mysql.plugin to '$CLONE_USER'@'localhost';
+ GRANT SELECT,UPDATE,INSERT ON performance_schema.* TO '$CLONE_USER'@'localhost';
+EOF
+
+    wsrep_log_info "Launching clone recipient daemon"
+    wsrep_log_info "-> using: $CLONE_ENV $CLONE_BINARY_SAFE $DEFAULT_OPTIONS "
+    wsrep_log_debug "-> Test connection as: -u$CLONE_USER -pxxxxxx -h $JOINER_CLONE_HOST -P $JOINER_CLONE_PORT"
+
+    # Define client to be used on the Joiner side
+    MYSQL_ACLIENT="$MYSQL_CLIENT -u$CLONE_USER -S$CLONE_SOCK --batch --skip_column_names --silent"
+    wsrep_log_debug "-> MYSQL_ACLIENT: $MYSQL_ACLIENT"
+
+    # HERE We start the instance that will get the clone and pass the CLONE_SQL file to create user
+    eval WSREP_SST_CLONE=1 $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS \
+        --wsrep_provider=none --init_file="$CLONE_SQL" >> $CLONE_ERR &
+
+    wsrep_log_info "-> Initialization user returned $?"
+
+    if [ ! "$?" == "0" ]; then
+        wsrep_log_error "-> User Creation on Joiner node failed, possible permission denied. Check permissions for "
+        #we will try to silently shutdown the instance
+        `$MYSQL_ACLIENT -e "SHUTDOWN" 2> /dev/null` || :
+        exit 1
+    fi
+
+    # Wait for the receiver process to start
+    until [ -n "$(cat $CLONE_PID_FILE 2>/dev/null)" ]
+    do
+        sleep 0.2
+    done
+    CLONE_INSTANCE_PID=`cat $CLONE_PID_FILE`
+
+    if [ -n "$WSREP_LOG_DEBUG" ]; then
+        GRANTS=`$MYSQL_ACLIENT -uroot -NB -e "show grants for '$CLONE_USER'@'localhost'"`
+        wsrep_log_debug "-> Clone user grants: $GRANTS"
```

Comment:
Do we really need this in our logs with wsrep-debug=1?
```
2025-02-11T07:45:25.787850Z 0 [Note] [MY-000000] [WSREP-SST] (debug) -> Clone user grants: GRANT SHUTDOWN, SUPER, EXECUTE ON *.* TO `clone_sst`@`localhost`
2025-02-11T07:45:25.787890Z 0 [Note] [MY-000000] [WSREP-SST] GRANT CLONE_ADMIN,SYSTEM_VARIABLES_ADMIN ON *.* TO `clone_sst`@`localhost`
2025-02-11T07:45:25.787898Z 0 [Note] [MY-000000] [WSREP-SST] GRANT SELECT, INSERT, UPDATE ON `performance_schema`.* TO `clone_sst`@`localhost`
2025-02-11T07:45:25.787905Z 0 [Note] [MY-000000] [WSREP-SST] GRANT INSERT ON `mysql`.`plugin` TO `clone_sst`@`localhost`
```

Reply:
for debug yes